<h1 style="text-align: center;"><a title="Data Prep-AIMS-Cmr-2024-25">Assignment 1, Data Prep  @AIMSSen 25</h1>

**Instructor:** 

* Rockefeller

**Name: WANDIYA JAMES**

**COOPERATIVE PROGRAM**

# **<span style="color:red">Score: 13.25 / 20 </span>**

# Exercice 1:  Trends in Business in Nigeria

**NaijaBiz Analytics** is a leading business intelligence firm that specializes in analyzing small and medium enterprises (SMEs) across Nigeria. The company has been contracted by the Nigerian Ministry of Industry, Trade and Investment to conduct a comprehensive analysis of business performance across different states and sectors. 

As a newly hired Data Analyst at NaijaBiz Analytics, you have been provided with a dataset containing information about businesses across various Nigerian states. Your task is to analyze the business data using basic filtering operations to provide actionable insights.

**Dataset:** `naija_businesses.csv`

For each question below, provide the Python pandas code and extract meaningful business insights from your analysis.


In [1]:
import pandas as pd 

In [2]:
# This is a dataset containing information about businesses across varoius Nigerian States
df_naija_businesses = pd.read_csv('/home/students-aimssn/Documents/Data_Prep/A1_data_prep/data/naija_businesses.csv')

In [3]:
df_naija_businesses

business_id business_name   state  business_type  start_date  \
0       NG_0001    Business_1   Benue         Retail  2023-02-01   
1       NG_0002    Business_2  Kaduna  Manufacturing  2023-11-17   
2       NG_0003    Business_3     Oyo     Technology  2020-09-02   
3       NG_0004    Business_4   Benue  Manufacturing  2023-04-14   
4       NG_0005    Business_5  Rivers     Technology  2022-11-28   
..          ...           ...     ...            ...         ...   
195     NG_0196  Business_196    Kano       Services  2023-09-27   
196     NG_0197  Business_197    Kano    Agriculture  2021-01-31   
197     NG_0198  Business_198  Kaduna    Agriculture  2021-11-29   
198     NG_0199  Business_199     Oyo    Agriculture  2022-11-08   
199     NG_0200  Business_200  Rivers         Retail  2020-06-25   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
0                   417741               406718          8        False   
1                   696804               256593         10         True   
2                   520713                87719         11        False   
3                   553527               389310          7        False   
4                   231537               -11869         10         True   
..                     ...                  ...        ...          ...   
195                 611128               415487          8         True   
196                 916895               349547          7        False   
197                 778180               430501          8        False   
198                  98486                78788          4        False   
199                 713491               339290         10        False   

     accepts_mobile_money  
0                    True  
1                   False  
2                    True  
3                    True  
4                   False  
..                    ...  
195                 False  
196                  True  
197                  True  
198                  True  
199                 False  

[200 rows x 10 columns]

### Q1: High Revenue Businesses
Filter and display all businesses that have monthly revenue greater than 600,000 Naira. How many such businesses exist?

**Business Context:** Identifying high-revenue businesses can help understand successful business models.


**<span style="color:red">Score:  1/ 1 </span>**

In [4]:
# This display businesses that have monthly revenue greater than 600,000 Naira
high_revenue_businesses = df_naija_businesses[df_naija_businesses['monthly_revenue_naira'] > 600000]


In [5]:
high_revenue_businesses

business_id business_name    state  business_type  start_date  \
1       NG_0002    Business_2   Kaduna  Manufacturing  2023-11-17   
6       NG_0007    Business_7      Oyo       Services  2021-06-21   
7       NG_0008    Business_8      Oyo     Technology  2022-07-09   
11      NG_0012   Business_12    Benue       Services  2022-06-21   
12      NG_0013   Business_13   Rivers  Manufacturing  2020-02-05   
18      NG_0019   Business_19  Anambra  Manufacturing  2021-10-15   
27      NG_0028   Business_28     Kano         Retail  2020-02-20   
29      NG_0030   Business_30   Kaduna         Retail  2023-04-28   
30      NG_0031   Business_31      Oyo       Services  2023-02-22   
36      NG_0037   Business_37   Kaduna       Services  2021-11-12   
38      NG_0039   Business_39    Lagos     Technology  2022-12-13   
39      NG_0040   Business_40   Rivers         Retail  2022-09-29   
40      NG_0041   Business_41   Rivers       Services  2021-03-19   
41      NG_0042   Business_42    Benue    Agriculture  2021-09-04   
43      NG_0044   Business_44  Anambra       Services  2020-09-20   
48      NG_0049   Business_49      Imo    Agriculture  2023-05-04   
54      NG_0055   Business_55    Benue     Technology  2020-02-16   
58      NG_0059   Business_59   Rivers         Retail  2021-10-17   
72      NG_0073   Business_73    Lagos       Services  2023-11-17   
76      NG_0077   Business_77   Kaduna    Agriculture  2021-05-27   
79      NG_0080   Business_80   Kaduna         Retail  2023-02-15   
83      NG_0084   Business_84  Anambra       Services  2022-06-15   
85      NG_0086   Business_86   Rivers  Manufacturing  2023-04-27   
87      NG_0088   Business_88    Lagos  Manufacturing  2021-08-03   
90      NG_0091   Business_91  Anambra       Services  2020-09-03   
93      NG_0094   Business_94      Imo  Manufacturing  2023-12-29   
94      NG_0095   Business_95      Imo         Retail  2022-09-23   
95      NG_0096   Business_96      Imo         Retail  2023-01-14   
106     NG_0107  Business_107      Imo    Agriculture  2023-05-12   
108     NG_0109  Business_109      Imo  Manufacturing  2020-08-05   
111     NG_0112  Business_112    Benue    Agriculture  2020-12-06   
114     NG_0115  Business_115    Benue     Technology  2023-07-01   
115     NG_0116  Business_116   Kaduna     Technology  2020-12-22   
118     NG_0119  Business_119  Anambra         Retail  2023-01-28   
119     NG_0120  Business_120      Oyo       Services  2022-03-12   
123     NG_0124  Business_124    Benue         Retail  2021-03-17   
124     NG_0125  Business_125      Oyo  Manufacturing  2022-02-07   
128     NG_0129  Business_129   Kaduna  Manufacturing  2022-11-05   
129     NG_0130  Business_130   Kaduna       Services  2022-03-02   
130     NG_0131  Business_131   Kaduna         Retail  2020-12-01   
137     NG_0138  Business_138    Lagos       Services  2021-01-14   
139     NG_0140  Business_140    Lagos         Retail  2021-11-21   
141     NG_0142  Business_142    Lagos         Retail  2023-07-15   
143     NG_0144  Business_144  Anambra         Retail  2023-04-06   
145     NG_0146  Business_146    Benue  Manufacturing  2020-10-11   
147     NG_0148  Business_148   Rivers         Retail  2020-01-12   
150     NG_0151  Business_151   Rivers       Services  2021-05-13   
157     NG_0158  Business_158     Kano     Technology  2021-08-27   
163     NG_0164  Business_164   Kaduna     Technology  2023-05-04   
173     NG_0174  Business_174   Kaduna       Services  2021-11-06   
176     NG_0177  Business_177   Rivers     Technology  2022-07-08   
179     NG_0180  Business_180   Rivers     Technology  2023-01-27   
183     NG_0184  Business_184   Rivers  Manufacturing  2021-10-24   
184     NG_0185  Business_185    Lagos     Technology  2023-04-28   
193     NG_0194  Business_194    Benue    Agriculture  2021-08-18   
194     NG_0195  Business_195    Benue    Agriculture  2021-12-29   
195     NG_0196  Business_196     Kano       Services  2023-09-27   
19

In [6]:
# This is to determine the high business model that understand successful business model
high_revenue_count = len(high_revenue_businesses)

In [7]:
# This shows the total number of high revenue businesses in percentage(100%)
high_revenue_count

60

### The Business insight: 

**The retail will emerges as a  dominant low-cost business model, because it demonstrates the sector to offer a vital opportunities for economic that has low capital demand**.

**These high-revenue businesses represent successful models that could be studied and replicated to help other SMEs scale up.**

### Q2: Technology Businesses with Websites
Filter businesses that are in the "Technology" sector AND have a website. Display their business names and monthly revenue.

**Business Context:** Understanding digital adoption in the tech sector can inform technology support programs.


**<span style="color:red">Score:  0.5/ 1 </span>**

In [8]:
# This is a dataset of business that in the technology
business_in_tech = df_naija_businesses[
    (df_naija_businesses['business_type'] == 'Technology') & (df_naija_businesses['has_website'] == True)
]

In [9]:
# This is a dataset of business that into technology as displayed below
business_in_tech

business_id business_name    state business_type  start_date  \
4       NG_0005    Business_5   Rivers    Technology  2022-11-28   
9       NG_0010   Business_10     Kano    Technology  2021-06-08   
33      NG_0034   Business_34     Kano    Technology  2021-07-23   
74      NG_0075   Business_75      Oyo    Technology  2022-12-17   
115     NG_0116  Business_116   Kaduna    Technology  2020-12-22   
144     NG_0145  Business_145  Anambra    Technology  2022-12-02   
161     NG_0162  Business_162    Lagos    Technology  2020-04-24   
176     NG_0177  Business_177   Rivers    Technology  2022-07-08   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
4                   231537               -11869         10         True   
9                   549862               293803          3         True   
33                  265604                 8259          8         True   
74                  370740               296592          6         True   
115                 742563               272479          8         True   
144                 349510               294444          8         True   
161                 319976               255980         12         True   
176                 673369               261477         10         True   

     accepts_mobile_money  
4                   False  
9                   False  
33                   True  
74                   True  
115                  True  
144                  True  
161                 False  
176                 False

In [10]:
business_tech_count= len(business_in_tech)

In [11]:
# when filtered I discover only eight buinesses only 
business_tech_count

8

In [12]:
percentage_business_tech = (business_tech_count /len(df_naija_businesses) * 100)

In [13]:
percentage_business_tech

4.0

### The Business insight: 

**The retail will emerges as a  dominant low-cost business model, because it demonstrates the sector to offer a vital opportunities for economic that has low capital demand**.

**The low website adoption rate (4%) among Technology businesses indicates a significant digital gap in Nigeria's tech sector. This alone presents an opportunity for technology support programs to focus on digital infrastructure development, specifically in helping tech businesses establish online presence to enhance their competitiveness and market reach**

### Q3: Manufacturing Businesses in Specific States
Filter all Manufacturing businesses located in either "Lagos" OR "Kano" states. How many manufacturing businesses are in these two states?

**Business Context:** These states are major industrial hubs, and this analysis helps understand manufacturing concentration.


**<span style="color:red">Score:  1/ 1 </span>**

In [14]:
specific_state_manufacturing = df_naija_businesses[
    (df_naija_businesses['business_type'] == 'Manufacturing') & 
    (df_naija_businesses['state'].isin(['Lagos', 'Kano']))
]

In [15]:
specific_state_manufacturing

business_id business_name  state  business_type  start_date  \
22      NG_0023   Business_23   Kano  Manufacturing  2023-01-20   
42      NG_0043   Business_43   Kano  Manufacturing  2022-09-19   
70      NG_0071   Business_71   Kano  Manufacturing  2022-03-01   
75      NG_0076   Business_76   Kano  Manufacturing  2021-09-25   
87      NG_0088   Business_88  Lagos  Manufacturing  2021-08-03   
92      NG_0093   Business_93   Kano  Manufacturing  2020-06-02   
105     NG_0106  Business_106   Kano  Manufacturing  2021-03-28   
122     NG_0123  Business_123   Kano  Manufacturing  2022-01-31   
127     NG_0128  Business_128  Lagos  Manufacturing  2021-05-30   
165     NG_0166  Business_166  Lagos  Manufacturing  2023-01-03   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
22                  589101               332379         11        False   
42                  279155               216038          4        False   
70                  435325               336302          8        False   
75                  419289               335431          6        False   
87                  746502               157898         14        False   
92                  533285               147207          8        False   
105                 414793                82169          6        False   
122                 572033               141804          5        False   
127                 123223                98578          6        False   
165                 288084               230467         11         True   

     accepts_mobile_money  
22                  False  
42                   True  
70                   True  
75                  False  
87                  False  
92                   True  
105                  True  
122                  True  
127                  True  
165                 False

In [16]:
manufactruing_states_count= len(specific_state_manufacturing)

In [17]:
manufactruing_states_count

10

In [18]:
percentange_manufactuuring_states = (manufactruing_states_count/len(df_naija_businesses) * 100)

### Business insight: 

**The data shows in Lagos and Kano are indeed the major industrial hub because it hosts over one-third of Nigeria's manufacturing businesses.**

 **The Kano shows stronger manufacturing concentration than Lagos.**
 **However, digital adoption remains low, indicating opportunities for technology integration in these traditional manufacturing hubs.**


### Q4: Small Businesses Analysis
Filter businesses that have fewer than 5 employees AND monthly costs less than 200,000 Naira. What business types do these small businesses represent?

**Business Context:** Small businesses often need different support strategies than larger enterprises.


**<span style="color:red">Score:  0.5/ 1 </span>**

In [19]:
small_business = df_naija_businesses[
    (df_naija_businesses['employees'] < 5) & (df_naija_businesses['monthly_costs_naira'] < 200000)
   ]

In [20]:
small_business

business_id business_name    state  business_type  start_date  \
16      NG_0017   Business_17   Kaduna         Retail  2021-05-26   
23      NG_0024   Business_24  Anambra       Services  2023-10-09   
30      NG_0031   Business_31      Oyo       Services  2023-02-22   
57      NG_0058   Business_58    Lagos       Services  2022-07-24   
71      NG_0072   Business_72     Kano       Services  2023-02-24   
83      NG_0084   Business_84  Anambra       Services  2022-06-15   
89      NG_0090   Business_90     Kano    Agriculture  2022-05-10   
123     NG_0124  Business_124    Benue         Retail  2021-03-17   
138     NG_0139  Business_139    Benue     Technology  2022-08-14   
153     NG_0154  Business_154   Rivers       Services  2021-12-27   
156     NG_0157  Business_157    Benue    Agriculture  2022-02-14   
181     NG_0182  Business_182    Benue  Manufacturing  2022-10-05   
198     NG_0199  Business_199      Oyo    Agriculture  2022-11-08   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
16                  212107               169685          2         True   
23                  207648               166118          4        False   
30                  795650                40562          3        False   
57                  306577                43908          4         True   
71                  167976                70428          1        False   
83                  805420               163626          4        False   
89                  403124               -29477          4        False   
123                 843719               190578          3        False   
138                  65023                52018          4        False   
153                 239679                13311          4        False   
156                  70828                56662          4         True   
181                 454650                88372          3        False   
198                  98486                78788          4        False   

     accepts_mobile_money  
16                  False  
23                   True  
30                   True  
57                   True  
71                   True  
83                   True  
89                   True  
123                  True  
138                 False  
153                  True  
156                 False  
181                  True  
198                  True

In [21]:
small_business_count = len(small_business)

In [22]:
small_business_count 

13

In [23]:
percent_result = (len(small_business)/len(df_naija_businesses)) * 100

In [24]:
print("The prercentage of small business is: ",round(percent_result, 2), "%" )  

The prercentage of small business is:  6.5 %


### The Business insight here: 

**From the above table, I undsertand it is a small scale business or micro type of business that meet up the demand or requirement.**
**They are about tweenty-three percent (6.5%).**

### Q5: Profitable Retail Businesses
Filter Retail businesses where monthly revenue is greater than monthly costs. Display their state and employee count.

**Business Context:** Understanding profitable retail operations can help identify successful business practices.


**<span style="color:red">Score:  1/ 1 </span>**

In [25]:
profitable_retail = df_naija_businesses[
    (df_naija_businesses['business_type']== 'Retail') & (df_naija_businesses['monthly_revenue_naira'] > df_naija_businesses['monthly_costs_naira'])
    ]

In [26]:
profitable_retail

business_id business_name    state business_type  start_date  \
0       NG_0001    Business_1    Benue        Retail  2023-02-01   
16      NG_0017   Business_17   Kaduna        Retail  2021-05-26   
26      NG_0027   Business_27      Imo        Retail  2022-05-03   
27      NG_0028   Business_28     Kano        Retail  2020-02-20   
28      NG_0029   Business_29  Anambra        Retail  2021-07-28   
29      NG_0030   Business_30   Kaduna        Retail  2023-04-28   
31      NG_0032   Business_32    Lagos        Retail  2020-05-29   
34      NG_0035   Business_35      Imo        Retail  2023-09-27   
37      NG_0038   Business_38    Lagos        Retail  2022-04-17   
39      NG_0040   Business_40   Rivers        Retail  2022-09-29   
44      NG_0045   Business_45   Kaduna        Retail  2022-12-12   
46      NG_0047   Business_47  Anambra        Retail  2020-01-05   
47      NG_0048   Business_48    Benue        Retail  2023-01-31   
53      NG_0054   Business_54   Kaduna        Retail  2023-10-05   
55      NG_0056   Business_56   Kaduna        Retail  2022-05-16   
56      NG_0057   Business_57  Anambra        Retail  2022-04-02   
58      NG_0059   Business_59   Rivers        Retail  2021-10-17   
63      NG_0064   Business_64    Lagos        Retail  2021-10-18   
64      NG_0065   Business_65    Benue        Retail  2020-04-29   
73      NG_0074   Business_74     Kano        Retail  2022-10-09   
78      NG_0079   Business_79    Benue        Retail  2023-01-24   
79      NG_0080   Business_80   Kaduna        Retail  2023-02-15   
94      NG_0095   Business_95      Imo        Retail  2022-09-23   
95      NG_0096   Business_96      Imo        Retail  2023-01-14   
97      NG_0098   Business_98   Kaduna        Retail  2023-08-01   
100     NG_0101  Business_101    Benue        Retail  2022-12-06   
101     NG_0102  Business_102     Kano        Retail  2022-10-05   
107     NG_0108  Business_108   Kaduna        Retail  2022-06-30   
110     NG_0111  Business_111  Anambra        Retail  2021-03-07   
117     NG_0118  Business_118      Imo        Retail  2021-06-05   
118     NG_0119  Business_119  Anambra        Retail  2023-01-28   
123     NG_0124  Business_124    Benue        Retail  2021-03-17   
126     NG_0127  Business_127     Kano        Retail  2023-09-10   
130     NG_0131  Business_131   Kaduna        Retail  2020-12-01   
132     NG_0133  Business_133    Lagos        Retail  2021-05-11   
139     NG_0140  Business_140    Lagos        Retail  2021-11-21   
141     NG_0142  Business_142    Lagos        Retail  2023-07-15   
143     NG_0144  Business_144  Anambra        Retail  2023-04-06   
147     NG_0148  Business_148   Rivers        Retail  2020-01-12   
148     NG_0149  Business_149    Lagos        Retail  2023-08-18   
158     NG_0159  Business_159    Lagos        Retail  2022-08-27   
160     NG_0161  Business_161    Benue        Retail  2021-02-21   
167     NG_0168  Business_168    Benue        Retail  2023-10-31   
168     NG_0169  Business_169    Benue        Retail  2023-12-14   
175     NG_0176  Business_176      Imo        Retail  2021-04-01   
178     NG_0179  Business_179    Lagos        Retail  2022-01-08   
182     NG_0183  Business_183      Imo        Retail  2020-04-05   
186     NG_0187  Business_187     Kano        Retail  2023-10-24   
189     NG_0190  Business_190      Imo        Retail  2022-05-16   
199     NG_0200  Business_200   Rivers        Retail  2020-06-25   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
0                   417741               406718          8        False   
16                  212107               169685          2         True   
26                  151342               121073         12        False   
27                  713895               168645         10        False   
28                  467226               270305          3         True   
29                  872155               371524          6        False   
31        

In [27]:
profitable_retail_count = len(profitable_retail)

In [28]:
profitable_retail_count

50

In [29]:
df_naija_businesses[['state','employees']]

state  employees
0     Benue          8
1    Kaduna         10
2       Oyo         11
3     Benue          7
4    Rivers         10
..      ...        ...
195    Kano          8
196    Kano          7
197  Kaduna          8
198     Oyo          4
199  Rivers         10

[200 rows x 2 columns]

### Q6: Businesses Accepting Mobile Money
Filter all businesses that accept mobile money AND have more than 8 employees. Which states are these businesses located in?

**Business Context:** Larger businesses adopting mobile payments indicate financial technology integration.


**<span style="color:red">Score:  0.75/ 1 </span>**

In [30]:
mobile_money_business = df_naija_businesses[
    (df_naija_businesses['accepts_mobile_money'] == True) & 
    (df_naija_businesses['employees'] > 8)
]

In [31]:
mobile_money_business  

business_id business_name    state  business_type  start_date  \
2       NG_0003    Business_3      Oyo     Technology  2020-09-02   
6       NG_0007    Business_7      Oyo       Services  2021-06-21   
10      NG_0011   Business_11   Rivers       Services  2022-04-07   
13      NG_0014   Business_14   Rivers    Agriculture  2021-11-15   
21      NG_0022   Business_22      Oyo  Manufacturing  2023-03-12   
26      NG_0027   Business_27      Imo         Retail  2022-05-03   
35      NG_0036   Business_36      Oyo       Services  2020-10-31   
36      NG_0037   Business_37   Kaduna       Services  2021-11-12   
37      NG_0038   Business_38    Lagos         Retail  2022-04-17   
41      NG_0042   Business_42    Benue    Agriculture  2021-09-04   
45      NG_0046   Business_46   Kaduna  Manufacturing  2022-09-30   
46      NG_0047   Business_47  Anambra         Retail  2020-01-05   
56      NG_0057   Business_57  Anambra         Retail  2022-04-02   
64      NG_0065   Business_65    Benue         Retail  2020-04-29   
66      NG_0067   Business_67   Kaduna    Agriculture  2022-12-17   
69      NG_0070   Business_70      Imo    Agriculture  2020-04-26   
73      NG_0074   Business_74     Kano         Retail  2022-10-09   
77      NG_0078   Business_78   Kaduna       Services  2022-03-12   
84      NG_0085   Business_85    Benue  Manufacturing  2023-07-19   
85      NG_0086   Business_86   Rivers  Manufacturing  2023-04-27   
93      NG_0094   Business_94      Imo  Manufacturing  2023-12-29   
97      NG_0098   Business_98   Kaduna         Retail  2023-08-01   
98      NG_0099   Business_99      Imo       Services  2023-05-28   
101     NG_0102  Business_102     Kano         Retail  2022-10-05   
110     NG_0111  Business_111  Anambra         Retail  2021-03-07   
119     NG_0120  Business_120      Oyo       Services  2022-03-12   
120     NG_0121  Business_121  Anambra       Services  2023-04-18   
135     NG_0136  Business_136      Oyo    Agriculture  2022-05-01   
136     NG_0137  Business_137    Lagos    Agriculture  2021-11-20   
141     NG_0142  Business_142    Lagos         Retail  2023-07-15   
155     NG_0156  Business_156     Kano       Services  2020-03-30   
158     NG_0159  Business_159    Lagos         Retail  2022-08-27   
163     NG_0164  Business_164   Kaduna     Technology  2023-05-04   
175     NG_0176  Business_176      Imo         Retail  2021-04-01   
177     NG_0178  Business_178   Rivers     Technology  2020-07-10   
179     NG_0180  Business_180   Rivers     Technology  2023-01-27   
180     NG_0181  Business_181      Oyo       Services  2022-09-04   
191     NG_0192  Business_192   Rivers    Agriculture  2023-07-20   
193     NG_0194  Business_194    Benue    Agriculture  2021-08-18   
194     NG_0195  Business_195    Benue    Agriculture  2021-12-29   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
2                   520713                87719         11        False   
6                   724708               207436         12        False   
10                  564903               326094         16        False   
13                  599212               367711         13        False   
21                  264774               211819          9         True   
26                  151342               121073         12        False   
35                  415564               334449         10        False   
36                  788749               212612         10        False   
37                  117018                93614         10         True   
41                  646634               369677         14        False   
45                    8161                 6528         16         True   
46                  511482               327526         11        False   
56                  359637               287709          9         True   
64                  592544               474035         13         True   
66                  229509               118424  

### Q7: Low-Cost Operations
Filter businesses where monthly costs are less than 150,000 Naira. What is the most common business type among these low-cost operations?

**Business Context:** Low-cost business models can be important for economic inclusion and job creation.


**<span style="color:red">Score:  0.5/ 1 IN MOST OF YOUR QUESTION YOU DONT HAVE ALL CORRECT BECAUSE YOU HAVENT READ WELL THE QUESTION </span>**

In [32]:
low_cost_businesses = df_naija_businesses[df_naija_businesses['monthly_costs_naira'] < 150000]

In [33]:
low_cost_businesses

business_id business_name    state  business_type  start_date  \
2       NG_0003    Business_3      Oyo     Technology  2020-09-02   
4       NG_0005    Business_5   Rivers     Technology  2022-11-28   
5       NG_0006    Business_6  Anambra     Technology  2022-01-28   
7       NG_0008    Business_8      Oyo     Technology  2022-07-09   
26      NG_0027   Business_27      Imo         Retail  2022-05-03   
30      NG_0031   Business_31      Oyo       Services  2023-02-22   
33      NG_0034   Business_34     Kano     Technology  2021-07-23   
34      NG_0035   Business_35      Imo         Retail  2023-09-27   
37      NG_0038   Business_38    Lagos         Retail  2022-04-17   
39      NG_0040   Business_40   Rivers         Retail  2022-09-29   
45      NG_0046   Business_46   Kaduna  Manufacturing  2022-09-30   
57      NG_0058   Business_58    Lagos       Services  2022-07-24   
65      NG_0066   Business_66     Kano       Services  2022-04-10   
66      NG_0067   Business_67   Kaduna    Agriculture  2022-12-17   
67      NG_0068   Business_68    Lagos     Technology  2023-10-01   
71      NG_0072   Business_72     Kano       Services  2023-02-24   
72      NG_0073   Business_73    Lagos       Services  2023-11-17   
77      NG_0078   Business_78   Kaduna       Services  2022-03-12   
89      NG_0090   Business_90     Kano    Agriculture  2022-05-10   
92      NG_0093   Business_93     Kano  Manufacturing  2020-06-02   
105     NG_0106  Business_106     Kano  Manufacturing  2021-03-28   
107     NG_0108  Business_108   Kaduna         Retail  2022-06-30   
112     NG_0113  Business_113  Anambra         Retail  2021-07-04   
122     NG_0123  Business_123     Kano  Manufacturing  2022-01-31   
124     NG_0125  Business_125      Oyo  Manufacturing  2022-02-07   
127     NG_0128  Business_128    Lagos  Manufacturing  2021-05-30   
132     NG_0133  Business_133    Lagos         Retail  2021-05-11   
133     NG_0134  Business_134      Oyo     Technology  2022-06-14   
136     NG_0137  Business_137    Lagos    Agriculture  2021-11-20   
137     NG_0138  Business_138    Lagos       Services  2021-01-14   
138     NG_0139  Business_139    Benue     Technology  2022-08-14   
140     NG_0141  Business_141  Anambra  Manufacturing  2023-12-21   
143     NG_0144  Business_144  Anambra         Retail  2023-04-06   
148     NG_0149  Business_149    Lagos         Retail  2023-08-18   
150     NG_0151  Business_151   Rivers       Services  2021-05-13   
151     NG_0152  Business_152   Rivers       Services  2022-01-23   
152     NG_0153  Business_153    Lagos     Technology  2020-02-04   
153     NG_0154  Business_154   Rivers       Services  2021-12-27   
156     NG_0157  Business_157    Benue    Agriculture  2022-02-14   
160     NG_0161  Business_161    Benue         Retail  2021-02-21   
164     NG_0165  Business_165     Kano       Services  2022-04-05   
178     NG_0179  Business_179    Lagos         Retail  2022-01-08   
181     NG_0182  Business_182    Benue  Manufacturing  2022-10-05   
182     NG_0183  Business_183      Imo         Retail  2020-04-05   
190     NG_0191  Business_191    Benue    Agriculture  2022-05-22   
191     NG_0192  Business_192   Rivers    Agriculture  2023-07-20   
198     NG_0199  Business_199      Oyo    Agriculture  2022-11-08   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
2                   520713                87719         11        False   
4                   231537               -11869         10         True   
5                  -172819              -138255          5        False   
7                   928641               111173         11        False   
26                  151342               121073         12        False   
30                  795650                40562          3        False   
33                  265604                 8259          8         True   
34                  182707                89967          5         True   
37              

In [34]:
total_low_cost = len(low_cost_businesses)

In [35]:
total_low_cost

47

In [36]:

print(f" The number of businesses with monthly costs that is less than ₦150,000 is : {total_low_cost}")

 The number of businesses with monthly costs that is less than ₦150,000 is : 47


In [37]:
print(f"Percentage of all businesses: {(total_low_cost/len(df_naija_businesses))*100:.1f}%")

Percentage of all businesses: 23.5%


### The Business insight here: 

**The retail will emerges as a  dominant low-cost business model, because it demonstrates the sector to offer a vital opportunities for economic that has low capital demand**.

### Q8: Recent Business Registrations
Filter businesses that started in 2023. Which business type is most common among these new businesses?

**Business Context:** Recent registration trends can indicate emerging business opportunities and economic growth sectors.


**<span style="color:red">Score:  0.75/ 1 </span>**

In [38]:
df_naija_businesses ['start_date'] = pd.to_datetime(df_naija_businesses['start_date'])

businesses_start_2023 = df_naija_businesses[
    df_naija_businesses['start_date'].dt.year == 2023]

In [39]:
businesses_start_2023

business_id business_name    state  business_type start_date  \
0       NG_0001    Business_1    Benue         Retail 2023-02-01   
1       NG_0002    Business_2   Kaduna  Manufacturing 2023-11-17   
3       NG_0004    Business_4    Benue  Manufacturing 2023-04-14   
21      NG_0022   Business_22      Oyo  Manufacturing 2023-03-12   
22      NG_0023   Business_23     Kano  Manufacturing 2023-01-20   
23      NG_0024   Business_24  Anambra       Services 2023-10-09   
29      NG_0030   Business_30   Kaduna         Retail 2023-04-28   
30      NG_0031   Business_31      Oyo       Services 2023-02-22   
34      NG_0035   Business_35      Imo         Retail 2023-09-27   
47      NG_0048   Business_48    Benue         Retail 2023-01-31   
48      NG_0049   Business_49      Imo    Agriculture 2023-05-04   
53      NG_0054   Business_54   Kaduna         Retail 2023-10-05   
62      NG_0063   Business_63      Oyo     Technology 2023-02-03   
67      NG_0068   Business_68    Lagos     Technology 2023-10-01   
71      NG_0072   Business_72     Kano       Services 2023-02-24   
72      NG_0073   Business_73    Lagos       Services 2023-11-17   
78      NG_0079   Business_79    Benue         Retail 2023-01-24   
79      NG_0080   Business_80   Kaduna         Retail 2023-02-15   
82      NG_0083   Business_83      Oyo     Technology 2023-06-14   
84      NG_0085   Business_85    Benue  Manufacturing 2023-07-19   
85      NG_0086   Business_86   Rivers  Manufacturing 2023-04-27   
91      NG_0092   Business_92   Kaduna       Services 2023-01-04   
93      NG_0094   Business_94      Imo  Manufacturing 2023-12-29   
95      NG_0096   Business_96      Imo         Retail 2023-01-14   
97      NG_0098   Business_98   Kaduna         Retail 2023-08-01   
98      NG_0099   Business_99      Imo       Services 2023-05-28   
106     NG_0107  Business_107      Imo    Agriculture 2023-05-12   
113     NG_0114  Business_114      Imo  Manufacturing 2023-08-05   
114     NG_0115  Business_115    Benue     Technology 2023-07-01   
118     NG_0119  Business_119  Anambra         Retail 2023-01-28   
120     NG_0121  Business_121  Anambra       Services 2023-04-18   
121     NG_0122  Business_122      Oyo       Services 2023-05-31   
126     NG_0127  Business_127     Kano         Retail 2023-09-10   
140     NG_0141  Business_141  Anambra  Manufacturing 2023-12-21   
141     NG_0142  Business_142    Lagos         Retail 2023-07-15   
142     NG_0143  Business_143   Kaduna  Manufacturing 2023-12-11   
143     NG_0144  Business_144  Anambra         Retail 2023-04-06   
148     NG_0149  Business_149    Lagos         Retail 2023-08-18   
149     NG_0150  Business_150  Anambra  Manufacturing 2023-06-30   
163     NG_0164  Business_164   Kaduna     Technology 2023-05-04   
165     NG_0166  Business_166    Lagos  Manufacturing 2023-01-03   
167     NG_0168  Business_168    Benue         Retail 2023-10-31   
168     NG_0169  Business_169    Benue         Retail 2023-12-14   
170     NG_0171  Business_171  Anambra  Manufacturing 2023-07-24   
174     NG_0175  Business_175  Anambra  Manufacturing 2023-11-18   
179     NG_0180  Business_180   Rivers     Technology 2023-01-27   
184     NG_0185  Business_185    Lagos     Technology 2023-04-28   
186     NG_0187  Business_187     Kano         Retail 2023-10-24   
191     NG_0192  Business_192   Rivers    Agriculture 2023-07-20   
195     NG_0196  Business_196     Kano       Services 2023-09-27   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
0                   417741               406718          8        False   
1                   696804               256593         10         True   
3                   553527               389310          7        False   
21                  264774               211819          9         True   
22                  589101               332379         11        False   
23                  207648               166118          4        False   
29        

In [40]:
businesses_start_2023_count = len(businesses_start_2023)

In [41]:
print(f"Percentage of all businesses: {(businesses_start_2023_count/len(df_naija_businesses))*100:.1f}%")

Percentage of all businesses: 25.0%


## Driven Business Context are:
**The retail will continue to dominate new business registrations in 2023.**
 **The fame of  entrepreneurial activity wil increase.**
  **The balanced mixture of business types can suggests different  economic growth rather than just the  considering a single sector.**

**There are some top pefoming business in 2023 such as retail in kanduna for busines_30 and few others**

### Q9: Large Employers
Filter businesses that have 10 or more employees. Calculate the average monthly revenue of these larger employers.

**Business Context:** Large employers are important for job creation and economic stability.


**<span style="color:red">Score:  0.5/ 1 </span>**

In [42]:
large_employers = df_naija_businesses [df_naija_businesses['employees'] >= 10]

In [43]:
large_employers

business_id business_name   state  business_type start_date  \
1       NG_0002    Business_2  Kaduna  Manufacturing 2023-11-17   
2       NG_0003    Business_3     Oyo     Technology 2020-09-02   
4       NG_0005    Business_5  Rivers     Technology 2022-11-28   
6       NG_0007    Business_7     Oyo       Services 2021-06-21   
7       NG_0008    Business_8     Oyo     Technology 2022-07-09   
..          ...           ...     ...            ...        ...   
186     NG_0187  Business_187    Kano         Retail 2023-10-24   
191     NG_0192  Business_192  Rivers    Agriculture 2023-07-20   
192     NG_0193  Business_193   Lagos       Services 2020-03-02   
194     NG_0195  Business_195   Benue    Agriculture 2021-12-29   
199     NG_0200  Business_200  Rivers         Retail 2020-06-25   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
1                   696804               256593         10         True   
2                   520713                87719         11        False   
4                   231537               -11869         10         True   
6                   724708               207436         12        False   
7                   928641               111173         11        False   
..                     ...                  ...        ...          ...   
186                 515053               412042         10        False   
191                 421793               141370         11         True   
192                 440378               209484         10        False   
194                 812562               504472         11        False   
199                 713491               339290         10        False   

     accepts_mobile_money  
1                   False  
2                    True  
4                   False  
6                    True  
7                   False  
..                    ...  
186                 False  
191                  True  
192                 False  
194                  True  
199                 False  

[62 rows x 10 columns]

In [44]:
large_employers_count   = len(large_employers)

In [45]:
print(f" The number of businesses that have 10 or more eployees is : {large_employers_count}")

 The number of businesses that have 10 or more eployees is : 62


In [46]:
print(f"The percentage of the businesses is : {(large_employers_count/len(df_naija_businesses))*100:.1f}%")

The percentage of the businesses is : 31.0%


### Business context insight:
**There are 31% of businesses that have more than 10 employees**

**Most of the business type here are manufcaturing, services, Technology, Agriclture, and retail** 

**This also indicate that; since there are different sectors, it increases the chance of employment and reduce joblesness in that communties leading to economic stability**


### Q10: Digital-Ready Businesses
Filter businesses that have a website AND accept mobile money. What percentage of the total businesses are digitally ready?

**Business Context:** Digital readiness indicates how prepared businesses are for the modern economy.


**<span style="color:red">Score:  1/ 1 </span>**

In [47]:
e_ready_business = df_naija_businesses [
    (df_naija_businesses['has_website'] == True ) & (df_naija_businesses['accepts_mobile_money'] == True)
]

In [48]:
e_ready_business

business_id business_name    state  business_type start_date  \
14      NG_0015   Business_15  Anambra    Agriculture 2020-01-20   
21      NG_0022   Business_22      Oyo  Manufacturing 2023-03-12   
28      NG_0029   Business_29  Anambra         Retail 2021-07-28   
33      NG_0034   Business_34     Kano     Technology 2021-07-23   
34      NG_0035   Business_35      Imo         Retail 2023-09-27   
37      NG_0038   Business_38    Lagos         Retail 2022-04-17   
40      NG_0041   Business_41   Rivers       Services 2021-03-19   
45      NG_0046   Business_46   Kaduna  Manufacturing 2022-09-30   
56      NG_0057   Business_57  Anambra         Retail 2022-04-02   
57      NG_0058   Business_58    Lagos       Services 2022-07-24   
58      NG_0059   Business_59   Rivers         Retail 2021-10-17   
64      NG_0065   Business_65    Benue         Retail 2020-04-29   
68      NG_0069   Business_69   Kaduna  Manufacturing 2022-01-10   
74      NG_0075   Business_75      Oyo     Technology 2022-12-17   
84      NG_0085   Business_85    Benue  Manufacturing 2023-07-19   
91      NG_0092   Business_92   Kaduna       Services 2023-01-04   
97      NG_0098   Business_98   Kaduna         Retail 2023-08-01   
108     NG_0109  Business_109      Imo  Manufacturing 2020-08-05   
115     NG_0116  Business_116   Kaduna     Technology 2020-12-22   
120     NG_0121  Business_121  Anambra       Services 2023-04-18   
132     NG_0133  Business_133    Lagos         Retail 2021-05-11   
144     NG_0145  Business_145  Anambra     Technology 2022-12-02   
158     NG_0159  Business_159    Lagos         Retail 2022-08-27   
174     NG_0175  Business_175  Anambra  Manufacturing 2023-11-18   
175     NG_0176  Business_176      Imo         Retail 2021-04-01   
178     NG_0179  Business_179    Lagos         Retail 2022-01-08   
191     NG_0192  Business_192   Rivers    Agriculture 2023-07-20   
193     NG_0194  Business_194    Benue    Agriculture 2021-08-18   

     monthly_revenue_naira  monthly_costs_naira  employees  has_website  \
14                  511114               422377          8         True   
21                  264774               211819          9         True   
28                  467226               270305          3         True   
33                  265604                 8259          8         True   
34                  182707                89967          5         True   
37                  117018                93614         10         True   
40                  610938               493260          8         True   
45                    8161                 6528         16         True   
56                  359637               287709          9         True   
57                  306577                43908          4         True   
58                  787350               224882          7         True   
64                  592544               474035         13         True   
68                  340352               187933          8         True   
74                  370740               296592          6         True   
84                  376842               301473         10         True   
91                  545047               436037          7         True   
97                  372552               279734          9         True   
108                 687414               259674          7         True   
115                 742563               272479          8         True   
120                 326283               228986          9         True   
132                 525608                66637          8         True   
144                 349510               294444          8         True   
158                 256302               205041          9         True   
174                 449416               359532          7         True   
175                 525486               318756         10         True   
178                 428358                56621          5         True   
191   

In [49]:
e_ready_business = len(e_ready_business)

print(f"Number of digitally-ready businesses (website + mobile money): {e_ready_business}")

Number of digitally-ready businesses (website + mobile money): 28


In [50]:
percentage_digitally_ready = (e_ready_business / len(df_naija_businesses)) * 100

In [51]:
print(f"Percentage of total businesses: {percentage_digitally_ready:.1f}%")

Percentage of total businesses: 14.0%


# Exercise 2: African Agriculture Analysis

**AgriTech Africa** is a research organization analyzing agricultural technology adoption across five African countries. You have been provided with a dataset containing information about farming communities and their technology usage patterns.

For each question below, provide the Python pandas code and provide a key insight you can get from your answer.


In [52]:
import pandas as pd

In [53]:
african_agirculture_info = pd.read_csv('/home/students-aimssn/Documents/Data_Prep/A1_data_prep/data/african_agriculture_dataset.csv')

In [54]:
african_agirculture_info

Community_Name       Country Crop_Type  Farm_Size  \
0            Mountain Cultivators Club  South Africa    Barley      39.96   
1             Prairie Cultivators Club  South Africa   Sorghum      77.62   
2                 Prairie Agri Society       Nigeria     Wheat      97.46   
3                Harvest Farmers Union       Nigeria     Maize      27.84   
4                 Valley Farmers Group         Ghana   Sorghum      70.60   
...                                ...           ...       ...        ...   
1995              Harvest Agri Society         Kenya     Maize      39.93   
1996            Mountain Farmers Group       Nigeria     Maize      50.99   
1997          Prairie Cultivators Club         Ghana      Rice      89.37   
1998  Sunrise Agricultural Association         Kenya     Wheat      87.01   
1999            Savannah Farmers Group       Nigeria   Sorghum      57.86   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
0                      40         Yes          SMS        318.05   
1                      48          No          SMS        704.85   
2                      14         Yes          SMS        439.41   
3                      55          No          SMS        668.23   
4                      53         Yes  Mobile Apps        801.16   
...                   ...         ...          ...           ...   
1995                    0         Yes          NaN        246.61   
1996                   90          No          NaN        205.38   
1997                   43          No  Mobile Apps        603.02   
1998                   99          No  Mobile Apps        370.42   
1999                   48          No  Mobile Apps        997.68   

     Fertilizer_Type Irrigation Weather_Station Annual_Rainfall  
0            Organic  Sprinkler             Yes         1046.21  
1            Organic       Drip              No          921.82  
2                NaN       Drip             Yes         1072.58  
3                NaN  Sprinkler              No          629.73  
4                NaN  Sprinkler              No          909.69  
...              ...        ...             ...             ...  
1995             NaN        NaN             Yes         1123.58  
1996             NaN       Drip             Yes          646.68  
1997         Organic       Drip              No          958.79  
1998       Inorganic  Sprinkler             Yes          677.78  
1999       Inorganic        NaN             Yes          748.18  

[2000 rows x 12 columns]

## Q1: Identifying Technology Leaders
**Business Question:** Which farming communities are leading in technology adoption, and how many of our partner communities have achieved a Technology Adoption Index above 80?

**Context:** Understanding technology leaders helps identify best practices for scaling across other communities.

**<span style="color:red">Score:  0/ 1 </span>**

In [55]:
techonology_lead = african_agirculture_info[african_agirculture_info['Tech_Adoption_Index'] >= 80]

In [56]:
techonology_lead_count  = len(techonology_lead)

In [57]:
techonology_lead_count

409

In [58]:
techonology_lead

Community_Name       Country Crop_Type  Farm_Size  \
8     Prairie Agricultural Association      Tanzania     Wheat      23.15   
11           Sunrise Growers Coalition  South Africa     Wheat      70.31   
16                  River Agri Society      Tanzania   Sorghum      74.55   
17               Savannah Agri Society  South Africa   Sorghum      14.50   
20           Savannah Cultivators Club         Kenya   Sorghum      46.27   
...                                ...           ...       ...        ...   
1991              Meadow Farmers Group         Kenya      Rice      84.23   
1993               Golden Agri Society         Kenya      Rice      90.02   
1994            Green Cultivators Club       Nigeria     Wheat      26.81   
1996            Mountain Farmers Group       Nigeria     Maize      50.99   
1998  Sunrise Agricultural Association         Kenya     Wheat      87.01   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
8                      86         Yes          SMS        504.69   
11                     85         Yes          SMS        409.37   
16                     83          No  Mobile Apps        691.11   
17                     91          No  Mobile Apps        398.23   
20                     92          No          SMS        686.81   
...                   ...         ...          ...           ...   
1991                   91         Yes          SMS        273.06   
1993                   80         Yes  Mobile Apps        664.49   
1994                   95          No          SMS        454.72   
1996                   90          No          NaN        205.38   
1998                   99          No  Mobile Apps        370.42   

     Fertilizer_Type Irrigation Weather_Station Annual_Rainfall  
8            Organic       Drip             Yes          621.79  
11           Organic  Sprinkler              No         1178.04  
16         Inorganic  Sprinkler             Yes          719.05  
17         Inorganic       Drip             Yes         1067.74  
20               NaN        NaN             Yes          612.64  
...              ...        ...             ...             ...  
1991       Inorganic  Sprinkler              No          658.16  
1993             NaN       Drip             Yes          710.92  
1994             NaN  Sprinkler             Yes          607.55  
1996             NaN       Drip             Yes          646.68  
1998       Inorganic  Sprinkler             Yes          677.78  

[409 rows x 12 columns]

In [59]:
techonology_lead['Community_Name']

8       Prairie Agricultural Association
11             Sunrise Growers Coalition
16                    River Agri Society
17                 Savannah Agri Society
20             Savannah Cultivators Club
                      ...               
1991                Meadow Farmers Group
1993                 Golden Agri Society
1994              Green Cultivators Club
1996              Mountain Farmers Group
1998    Sunrise Agricultural Association
Name: Community_Name, Length: 409, dtype: object

## Q2: Premium Drone Adoption Analysis
**Business Question:** Among large-scale farming operations (over 70 acres), which communities have invested in drone technology, and where are they located?

**Context:** Large farms with drone adoption represent high-value clients for advanced agricultural technology partnerships.

**<span style="color:red">Score:  0.25/ 1 </span>**

In [60]:
analyze_large_farm = african_agirculture_info [(african_agirculture_info['Farm_Size'] > 70) &
    african_agirculture_info ['Drone_Usage']==True]

In [61]:
analyze_large_farm

Community_Name       Country Crop_Type  Farm_Size  \
1             Prairie Cultivators Club  South Africa   Sorghum      77.62   
2                 Prairie Agri Society       Nigeria     Wheat      97.46   
4                 Valley Farmers Group         Ghana   Sorghum      70.60   
5                Savannah Agri Society      Tanzania    Barley      92.92   
11           Sunrise Growers Coalition  South Africa     Wheat      70.31   
...                                ...           ...       ...        ...   
1988         Savannah Cultivators Club         Ghana     Wheat      97.13   
1991              Meadow Farmers Group         Kenya      Rice      84.23   
1993               Golden Agri Society         Kenya      Rice      90.02   
1997          Prairie Cultivators Club         Ghana      Rice      89.37   
1998  Sunrise Agricultural Association         Kenya     Wheat      87.01   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
1                      48          No          SMS        704.85   
2                      14         Yes          SMS        439.41   
4                      53         Yes  Mobile Apps        801.16   
5                      30         Yes          SMS        784.90   
11                     85         Yes          SMS        409.37   
...                   ...         ...          ...           ...   
1988                   29         Yes  Mobile Apps        555.20   
1991                   91         Yes          SMS        273.06   
1993                   80         Yes  Mobile Apps        664.49   
1997                   43          No  Mobile Apps        603.02   
1998                   99          No  Mobile Apps        370.42   

     Fertilizer_Type Irrigation Weather_Station Annual_Rainfall  
1            Organic       Drip              No          921.82  
2                NaN       Drip             Yes         1072.58  
4                NaN  Sprinkler              No          909.69  
5            Organic  Sprinkler              No          872.48  
11           Organic  Sprinkler              No         1178.04  
...              ...        ...             ...             ...  
1988       Inorganic  Sprinkler              No          663.25  
1991       Inorganic  Sprinkler              No          658.16  
1993             NaN       Drip             Yes          710.92  
1997         Organic       Drip              No          958.79  
1998       Inorganic  Sprinkler             Yes          677.78  

[669 rows x 12 columns]

## Q3: Market Expansion Opportunities
**Business Question:** How many farming communities are operating in our priority markets of South Africa and Kenya?

**Context:** These countries represent key expansion opportunities for agricultural technology solutions.

**<span style="color:red">Score:  1/ 1 </span>**

In [62]:
priority_markets = african_agirculture_info[african_agirculture_info['Country'].isin(['South Africa', 'Kenya'])]

In [63]:
priority_markets

Community_Name       Country Crop_Type  Farm_Size  \
0            Mountain Cultivators Club  South Africa    Barley      39.96   
1             Prairie Cultivators Club  South Africa   Sorghum      77.62   
6              River Growers Coalition  South Africa   Sorghum      68.94   
9            Sunrise Growers Coalition         Kenya     Maize      53.98   
11           Sunrise Growers Coalition  South Africa     Wheat      70.31   
...                                ...           ...       ...        ...   
1989              Harvest Agri Society         Kenya      Rice      55.40   
1991              Meadow Farmers Group         Kenya      Rice      84.23   
1993               Golden Agri Society         Kenya      Rice      90.02   
1995              Harvest Agri Society         Kenya     Maize      39.93   
1998  Sunrise Agricultural Association         Kenya     Wheat      87.01   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
0                      40         Yes          SMS        318.05   
1                      48          No          SMS        704.85   
6                      60          No          NaN        437.78   
9                      59         Yes          NaN        883.72   
11                     85         Yes          SMS        409.37   
...                   ...         ...          ...           ...   
1989                   24          No          SMS        658.68   
1991                   91         Yes          SMS        273.06   
1993                   80         Yes  Mobile Apps        664.49   
1995                    0         Yes          NaN        246.61   
1998                   99          No  Mobile Apps        370.42   

     Fertilizer_Type Irrigation Weather_Station Annual_Rainfall  
0            Organic  Sprinkler             Yes         1046.21  
1            Organic       Drip              No          921.82  
6          Inorganic        NaN             Yes          897.94  
9            Organic        NaN              No          629.33  
11           Organic  Sprinkler              No         1178.04  
...              ...        ...             ...             ...  
1989             NaN  Sprinkler             Yes          693.57  
1991       Inorganic  Sprinkler              No          658.16  
1993             NaN       Drip             Yes          710.92  
1995             NaN        NaN             Yes         1123.58  
1998       Inorganic  Sprinkler             Yes          677.78  

[760 rows x 12 columns]

In [64]:
total_prority_markets = len(priority_markets)

In [65]:
total_prority_markets

760

## Q4: Underserved Market Identification
**Business Question:** Which countries have small-scale, low-technology farming operations (under 20 acres with Technology Adoption Index below 30) that could benefit from basic technology interventions?

**Context:** Small, low-tech farms represent an underserved market segment requiring tailored entry-level solutions.


**<span style="color:red">Score:  1/ 1 </span>**

In [66]:
low_tech_communities = african_agirculture_info[
    (african_agirculture_info['Farm_Size'] < 20) & (african_agirculture_info['Tech_Adoption_Index'] < 30)
   ]

In [67]:
low_tech_communities

Community_Name       Country Crop_Type  Farm_Size  \
15                 Meadow Agri Society       Nigeria    Barley       8.72   
41     Valley Agricultural Association       Nigeria      Rice       2.16   
53              River Cultivators Club  South Africa    Barley       9.59   
54    Sunrise Agricultural Association  South Africa     Wheat       3.14   
81          Mountain Growers Coalition         Kenya      Rice      13.85   
...                                ...           ...       ...        ...   
1919         Mountain Cultivators Club      Tanzania   Sorghum      15.37   
1957         Prairie Growers Coalition       Nigeria    Barley      16.16   
1976         Mountain Cultivators Club         Ghana     Wheat      18.76   
1978               Meadow Agri Society         Kenya     Wheat       2.19   
1990              Golden Farmers Union       Nigeria     Wheat      19.07   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
15                     28          No          SMS        607.82   
41                     28         Yes          SMS        232.53   
53                     28         Yes          SMS        255.60   
54                     10          No          NaN        531.49   
81                      1          No          NaN        801.18   
...                   ...         ...          ...           ...   
1919                   17          No          SMS        786.98   
1957                   25          No  Mobile Apps        834.51   
1976                   16          No  Mobile Apps        781.23   
1978                    2         Yes          NaN        112.96   
1990                    6          No          SMS        904.24   

     Fertilizer_Type Irrigation Weather_Station Annual_Rainfall  
15           Organic       Drip             Yes          428.85  
41               NaN  Sprinkler              No          522.66  
53           Organic        NaN              No          813.43  
54         Inorganic  Sprinkler             Yes          405.82  
81               NaN  Sprinkler              No         1007.81  
...              ...        ...             ...             ...  
1919             NaN  Sprinkler             Yes          581.72  
1957             NaN       Drip             Yes          630.79  
1976       Inorganic       Drip             Yes          463.61  
1978       Inorganic        NaN             Yes          643.36  
1990             NaN        NaN              No          736.37  

[113 rows x 12 columns]

In [68]:
total_tech_communities = len(low_tech_communities)

In [69]:
total_tech_communities

113

## Q5: Sustainable Agriculture Success Stories
**Business Question:** Which organic farming communities are achieving high productivity (over 600 annual yield), and what irrigation technologies are they using?

**Context:** Successful organic farms demonstrate sustainable practices that can be promoted to environmentally conscious markets.

**<span style="color:red">Score:  1/ 1 </span>**

In [70]:
achieved_organic = african_agirculture_info[
    (african_agirculture_info['Fertilizer_Type'] == 'Organic') & (african_agirculture_info['Annual_Yield'] > 600)
    ]

In [71]:
achieved_organic

Community_Name       Country Crop_Type  Farm_Size  \
1             Prairie Cultivators Club  South Africa   Sorghum      77.62   
5                Savannah Agri Society      Tanzania    Barley      92.92   
7               Savannah Farmers Union       Nigeria      Rice      68.43   
9            Sunrise Growers Coalition         Kenya     Maize      53.98   
15                 Meadow Agri Society       Nigeria    Barley       8.72   
...                                ...           ...       ...        ...   
1959   Meadow Agricultural Association      Tanzania   Sorghum      65.94   
1963          Valley Growers Coalition  South Africa   Sorghum      36.55   
1964               Green Farmers Union         Kenya    Barley      13.11   
1973  Harvest Agricultural Association       Nigeria     Wheat      64.74   
1997          Prairie Cultivators Club         Ghana      Rice      89.37   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
1                      48          No          SMS        704.85   
5                      30         Yes          SMS        784.90   
7                      57          No          NaN        611.11   
9                      59         Yes          NaN        883.72   
15                     28          No          SMS        607.82   
...                   ...         ...          ...           ...   
1959                   78          No          SMS        756.00   
1963                   15         Yes          NaN        735.03   
1964                   78         Yes          NaN        974.90   
1973                   30          No          SMS        745.33   
1997                   43          No  Mobile Apps        603.02   

     Fertilizer_Type Irrigation Weather_Station Annual_Rainfall  
1            Organic       Drip              No          921.82  
5            Organic  Sprinkler              No          872.48  
7            Organic       Drip              No          534.91  
9            Organic        NaN              No          629.33  
15           Organic       Drip             Yes          428.85  
...              ...        ...             ...             ...  
1959         Organic  Sprinkler             Yes          574.38  
1963         Organic       Drip              No         1087.69  
1964         Organic  Sprinkler             Yes          950.74  
1973         Organic  Sprinkler              No          653.33  
1997         Organic       Drip              No          958.79  

[277 rows x 12 columns]

In [72]:
total_achieved_organic = len(achieved_organic)

In [73]:
total_achieved_organic

277

## Q6: Climate-Smart Agriculture Adoption
**Business Question:** In high-rainfall regions (over 1000mm annually), which countries have farming communities that have invested in weather monitoring technology?

**Context:** Weather station adoption in high-rainfall areas indicates sophisticated climate risk management practices.


**<span style="color:red">Score:  0.25/ 1 </span>**

In [74]:

african_agirculture_info['Annual_Rainfall'] = pd.to_numeric(african_agirculture_info['Annual_Rainfall'], errors = 'coerce')

high_rainfall_regions =  african_agirculture_info[african_agirculture_info['Annual_Rainfall'] > 1000]

In [75]:
high_rainfall_regions

Community_Name       Country Crop_Type  Farm_Size  \
0           Mountain Cultivators Club  South Africa    Barley      39.96   
2                Prairie Agri Society       Nigeria     Wheat      97.46   
11          Sunrise Growers Coalition  South Africa     Wheat      70.31   
12    Valley Agricultural Association         Ghana    Barley      99.69   
17              Savannah Agri Society  South Africa   Sorghum      14.50   
...                               ...           ...       ...        ...   
1970           River Cultivators Club         Ghana     Wheat      54.14   
1971         Sunrise Cultivators Club  South Africa   Sorghum      65.22   
1981              Meadow Agri Society         Kenya     Wheat      32.83   
1987               Green Agri Society         Kenya    Barley      20.11   
1995             Harvest Agri Society         Kenya     Maize      39.93   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
0                      40         Yes          SMS        318.05   
2                      14         Yes          SMS        439.41   
11                     85         Yes          SMS        409.37   
12                     60          No  Mobile Apps        182.56   
17                     91          No  Mobile Apps        398.23   
...                   ...         ...          ...           ...   
1970                    2          No  Mobile Apps        826.23   
1971                   72          No          NaN        526.41   
1981                   54         Yes          NaN        416.44   
1987                   39         Yes          SMS        431.03   
1995                    0         Yes          NaN        246.61   

     Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
0            Organic  Sprinkler             Yes          1046.21  
2                NaN       Drip             Yes          1072.58  
11           Organic  Sprinkler              No          1178.04  
12           Organic       Drip             Yes          1163.25  
17         Inorganic       Drip             Yes          1067.74  
...              ...        ...             ...              ...  
1970             NaN        NaN             Yes          1176.19  
1971       Inorganic        NaN              No          1052.29  
1981       Inorganic        NaN              No          1017.13  
1987         Organic       Drip             Yes          1081.81  
1995             NaN        NaN             Yes          1123.58  

[467 rows x 12 columns]

## Q7: Entry-Level Market Assessment
**Business Question:** Among low-performing farming operations (annual yield under 200), which crop types are most prevalent and may need immediate intervention?

**Context:** Low-yield operations represent potential clients for productivity improvement solutions.

**<span style="color:red">Score:  0.75/ 1  OU DIDNT ANSWER THE QUESTION</span>**

In [76]:
low_performing_farming = african_agirculture_info[african_agirculture_info['Annual_Yield'] < 200]

In [77]:
low_performing_farming

Community_Name   Country Crop_Type  Farm_Size  \
12    Valley Agricultural Association     Ghana    Barley      99.69   
30              Harvest Farmers Union   Nigeria   Sorghum       5.88   
39    Meadow Agricultural Association     Ghana      Rice      16.93   
44             River Cultivators Club     Ghana    Barley      31.74   
55              Mountain Agri Society   Nigeria     Wheat      59.81   
...                               ...       ...       ...        ...   
1950  Valley Agricultural Association  Tanzania     Wheat      43.91   
1965           Mountain Farmers Group  Tanzania     Maize      22.47   
1966          Valley Cultivators Club     Ghana      Rice      86.62   
1978              Meadow Agri Society     Kenya     Wheat       2.19   
1992             Valley Farmers Union  Tanzania    Barley      40.12   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
12                     60          No  Mobile Apps        182.56   
30                     83         Yes  Mobile Apps        136.35   
39                     71          No          NaN        181.94   
44                     46         Yes  Mobile Apps        198.41   
55                     52          No          SMS        154.01   
...                   ...         ...          ...           ...   
1950                    9          No          SMS        150.15   
1965                   88         Yes          SMS        139.24   
1966                   96          No          SMS        184.45   
1978                    2         Yes          NaN        112.96   
1992                   49         Yes          SMS        178.31   

     Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
12           Organic       Drip             Yes          1163.25  
30               NaN       Drip             Yes          1024.79  
39           Organic        NaN             Yes          1052.89  
44           Organic       Drip             Yes           498.94  
55           Organic       Drip             Yes          1137.86  
...              ...        ...             ...              ...  
1950       Inorganic       Drip             Yes           519.95  
1965         Organic  Sprinkler              No          1100.21  
1966             NaN        NaN              No           483.89  
1978       Inorganic        NaN             Yes           643.36  
1992             NaN       Drip             Yes           845.96  

[237 rows x 12 columns]

In [78]:
total_low_peforming_farming = len(low_performing_farming)

In [79]:
total_low_peforming_farming

237

## Q8: Digital Agriculture Penetration
**Business Question:** What irrigation methods are most commonly used by digitally-enabled farming communities (those using Mobile Apps or SMS technology)?

**Context:** Understanding technology adoption patterns helps design integrated digital agriculture solutions.

**<span style="color:red">Score:  0.5/ 1 </span>**

In [80]:
digital_agriculture = african_agirculture_info[african_agirculture_info['Mobile_Tech'].isin(['Mobile Apps','SMS'])]

In [81]:
digital_agriculture

Community_Name       Country Crop_Type  Farm_Size  \
0            Mountain Cultivators Club  South Africa    Barley      39.96   
1             Prairie Cultivators Club  South Africa   Sorghum      77.62   
2                 Prairie Agri Society       Nigeria     Wheat      97.46   
3                Harvest Farmers Union       Nigeria     Maize      27.84   
4                 Valley Farmers Group         Ghana   Sorghum      70.60   
...                                ...           ...       ...        ...   
1993               Golden Agri Society         Kenya      Rice      90.02   
1994            Green Cultivators Club       Nigeria     Wheat      26.81   
1997          Prairie Cultivators Club         Ghana      Rice      89.37   
1998  Sunrise Agricultural Association         Kenya     Wheat      87.01   
1999            Savannah Farmers Group       Nigeria   Sorghum      57.86   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
0                      40         Yes          SMS        318.05   
1                      48          No          SMS        704.85   
2                      14         Yes          SMS        439.41   
3                      55          No          SMS        668.23   
4                      53         Yes  Mobile Apps        801.16   
...                   ...         ...          ...           ...   
1993                   80         Yes  Mobile Apps        664.49   
1994                   95          No          SMS        454.72   
1997                   43          No  Mobile Apps        603.02   
1998                   99          No  Mobile Apps        370.42   
1999                   48          No  Mobile Apps        997.68   

     Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
0            Organic  Sprinkler             Yes          1046.21  
1            Organic       Drip              No           921.82  
2                NaN       Drip             Yes          1072.58  
3                NaN  Sprinkler              No           629.73  
4                NaN  Sprinkler              No           909.69  
...              ...        ...             ...              ...  
1993             NaN       Drip             Yes           710.92  
1994             NaN  Sprinkler             Yes           607.55  
1997         Organic       Drip              No           958.79  
1998       Inorganic  Sprinkler             Yes           677.78  
1999       Inorganic        NaN             Yes           748.18  

[1288 rows x 12 columns]

In [82]:
total_digital_agriculture = len(digital_agriculture)

In [83]:
total_digital_agriculture

1288

In [84]:
digital_agriculture_percentage = (total_digital_agriculture /len(african_agirculture_info) * 100)

In [85]:
print(f"Percentage of digital agriculture is : {digital_agriculture_percentage:.2f}%")

Percentage of digital agriculture is : 64.40%


## Q9: Large-Scale Technology Investment Patterns
**Business Question:** What is the average technology adoption level among large-scale farming operations (90+ acres), and how does it compare to smaller operations?

**Context:** Large farms typically have higher investment capacity and can be early adopters of expensive technologies.

**<span style="color:red">Score:  0/ 1 READ WELL THE QUESTION </span>**

In [86]:
analyze_large_scale_tech = african_agirculture_info[african_agirculture_info['Tech_Adoption_Index'] >= 90]

In [87]:
analyze_large_scale_tech

Community_Name       Country Crop_Type  Farm_Size  \
17               Savannah Agri Society  South Africa   Sorghum      14.50   
20           Savannah Cultivators Club         Kenya   Sorghum      46.27   
36              Mountain Farmers Union  South Africa     Wheat       7.32   
49                 Meadow Agri Society         Ghana   Sorghum      19.84   
59            Meadow Growers Coalition       Nigeria     Maize       9.96   
...                                ...           ...       ...        ...   
1983               Golden Agri Society  South Africa   Sorghum      89.55   
1991              Meadow Farmers Group         Kenya      Rice      84.23   
1994            Green Cultivators Club       Nigeria     Wheat      26.81   
1996            Mountain Farmers Group       Nigeria     Maize      50.99   
1998  Sunrise Agricultural Association         Kenya     Wheat      87.01   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
17                     91          No  Mobile Apps        398.23   
20                     92          No          SMS        686.81   
36                     91          No  Mobile Apps        377.96   
49                     91         Yes  Mobile Apps        995.02   
59                     95         Yes  Mobile Apps        818.11   
...                   ...         ...          ...           ...   
1983                   99          No          NaN        701.67   
1991                   91         Yes          SMS        273.06   
1994                   95          No          SMS        454.72   
1996                   90          No          NaN        205.38   
1998                   99          No  Mobile Apps        370.42   

     Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
17         Inorganic       Drip             Yes          1067.74  
20               NaN        NaN             Yes           612.64  
36               NaN       Drip             Yes           484.47  
49           Organic        NaN             Yes           775.61  
59         Inorganic       Drip             Yes          1016.78  
...              ...        ...             ...              ...  
1983             NaN       Drip              No           526.19  
1991       Inorganic  Sprinkler              No           658.16  
1994             NaN  Sprinkler             Yes           607.55  
1996             NaN       Drip             Yes           646.68  
1998       Inorganic  Sprinkler             Yes           677.78  

[212 rows x 12 columns]

In [88]:
average_large_scale_tech = analyze_large_scale_tech['Tech_Adoption_Index'].mean()

In [89]:
print(f"The average technology adoption level among large-scale farming operations is:", round(average_large_scale_tech, 1))

The average technology adoption level among large-scale farming operations is: 95.0



## Q10: Natural Farming Excellence
**Business Question:** What percentage of farming communities achieve high productivity (over 500 annual yield) without using fertilizers, demonstrating natural farming excellence?

**Context:** High-performing farms without fertilizer inputs represent efficient operations that could mentor others in sustainable practices.


**<span style="color:red">Score:  1/ 1 </span>**

In [90]:
analyze_natural_farming = african_agirculture_info[
    (african_agirculture_info['Fertilizer_Type'].isna()) & 
    (african_agirculture_info['Annual_Yield'] > 500)
]

In [91]:
analyze_natural_farming

Community_Name       Country Crop_Type  Farm_Size  \
3         Harvest Farmers Union       Nigeria     Maize      27.84   
4          Valley Farmers Group         Ghana   Sorghum      70.60   
18        Mountain Agri Society       Nigeria     Wheat      62.62   
19         Valley Farmers Group         Kenya     Maize      33.14   
20    Savannah Cultivators Club         Kenya   Sorghum      46.27   
...                         ...           ...       ...        ...   
1983        Golden Agri Society  South Africa   Sorghum      89.55   
1984       Meadow Farmers Union         Ghana     Wheat      84.12   
1989       Harvest Agri Society         Kenya      Rice      55.40   
1990       Golden Farmers Union       Nigeria     Wheat      19.07   
1993        Golden Agri Society         Kenya      Rice      90.02   

      Tech_Adoption_Index Drone_Usage  Mobile_Tech  Annual_Yield  \
3                      55          No          SMS        668.23   
4                      53         Yes  Mobile Apps        801.16   
18                     26         Yes          NaN        700.75   
19                     43          No  Mobile Apps        943.97   
20                     92          No          SMS        686.81   
...                   ...         ...          ...           ...   
1983                   99          No          NaN        701.67   
1984                   39          No          SMS        845.32   
1989                   24          No          SMS        658.68   
1990                    6          No          SMS        904.24   
1993                   80         Yes  Mobile Apps        664.49   

     Fertilizer_Type Irrigation Weather_Station  Annual_Rainfall  
3                NaN  Sprinkler              No           629.73  
4                NaN  Sprinkler              No           909.69  
18               NaN  Sprinkler             Yes          1132.60  
19               NaN        NaN              No           589.20  
20               NaN        NaN             Yes           612.64  
...              ...        ...             ...              ...  
1983             NaN       Drip              No           526.19  
1984             NaN       Drip              No           901.27  
1989             NaN  Sprinkler             Yes           693.57  
1990             NaN        NaN              No           736.37  
1993             NaN       Drip             Yes           710.92  

[374 rows x 12 columns]

In [92]:
total_analyze_natural_farming = len(analyze_natural_farming)

In [93]:
total_analyze_natural_farming

374

In [94]:
analyze_natural_farming_percentage = (total_analyze_natural_farming /len(african_agirculture_info) * 100)

In [95]:
analyze_natural_farming_percentage

18.7

### Context insight drived: ###
**The percentage of all the communities that achieve excellence naturally  is 18.7**

**Success rate of natural farming (what % of natural farms are high performers)**

**Geographic distribution to identify regions with successful natural farming practices**

**Yield comparisons to see if natural excellence farms compete with fertilized farms**